2D Simulation/movesCorrector.js

```diff
@@ -103,4 +95,32 @@ let movesCorrector = new function () {
     function is_valid_id(id) {
         return id % 1 === 0 && id >= 0 && id < NumBlocks;
     }
+
+    this.is_handling_ambiguity = function() {
+        return handling_ambiguity;
+    };
+
+    this.handle_ambiguity = function(currMove) {
+        console.log('handling ambiguity');
+        handling_ambiguity = true;
+        move = currMove;
+        blocks.display_block_ids();
+    };
+
+    this.finish_handling_ambiguity = function(message) {
+        console.log('this should be doing it.');
+        let id = convert_message_to_id_num(message);
+        move.block_id = 'block' + id;
+
+        handling_ambiguity = false;
+        blocks.display_block_letters();
+
+        if (move.type == 'ambiguous_move') {
+            move.type = 'move';
+            update_position(move);
+        } else if (move.type == 'ambiguous_flip') {
+            move.type = 'flip';
```

Comment:
This doesn't look like it's used. `update_flip_block` takes `move.block_id`, not `move`.

Reply:
It isn't, but having a move type equal to `"ambiguous_flip"` when it is no longer ambiguous seems bad to me.